### DISCUSSION TITLE
Run command in "main" tab when child tab closes

### DISCUSSION THREAD

**Discussion by tengl** [kovidgoyal/kitty#5021] [Q&A] [answered]
Hi! 
I need help with a script. I have this alias:

```shell
alias jrnlw='f(){ kitty @ launch zsh -c "jrnl work $@ --edit" --watcher ~/.dotfiles/kitty/jrnl_last.py };f'
```
Which opens `jrnl` in a new tab and the tab closes when the edit is done. After the tab is closed I would like to display the last journal entry in the "main" tab/window (where I ran the alias). I have this Python script as sent to the watcher above, but I don't know Python so I need help (this is probably the first line of python I've ever written so I don't really know what to do). I guess the command runs in the closing tab, not the main tab. 

```python
import os

def on_close(boss, window, data):
  os.system("jrnl work --format markdown -n 1 | glow -")
```

_Kitty 0.19.3_

**Comment by kovidgoyal** [comment, marked as answer]
You have two separate windows. The easiest way to co-ordinate with them is have the second window run a shell script that runs the journal command to edit the journal. Once that command exists it should use kitty @ send-text to send the command to display the journal to the first window. In order to do that the script should be called with the value of $KITTY_WINDOW_ID of the first window, which you can supply in the kitty @ launch command you run in the first window. No need to use a watcher at all, just  the remote control API

**Reply by tengl** [reply]
I see, thanks for the quick reply, I will try that.

**Reply by tengl** [reply]
So here's the result.

The alias

```shell
alias jrnlw='kitty @ launch zsh -c "~/.dotfiles/kitty/jrnlw.sh $KITTY_WINDOW_ID"'
```

jrnlw.sh

```shell
#!/bin/bash

jrnl work --edit "now:"

if ([[ ! -z "$1" ]]) then
  echo " jrnl work -n 1 --format markdown | glow -" | kitty @ send-text --match id:$1 --stdin 
fi
```
